src/ray/core_worker/context.cc

```diff
@@ -303,15 +328,14 @@ void WorkerContext::SetCurrentTask(const TaskSpecification &task_spec) {
   } else {
     RAY_CHECK(false);
   }
+
   if (task_spec.IsNormalTask() || task_spec.IsActorCreationTask()) {
-    // TODO(architkulkarni): Once workers are cached by runtime env, we should
-    // only set runtime_env_ once and then RAY_CHECK that we
-    // never see a new one.
```

Comment:
Actually, workers are already cached by runtime env which means a single worker only runs tasks with the same runtime env, so we don't need a cache but just add a RAY_CHECK that we never see a new one.

Reply:
OK